pytorch_lightning/utilities/boring_model.py

```diff
@@ -0,0 +1,203 @@
+# Copyright The PyTorch Lightning team.
```

Comment:
note for reviewing. Adding boring_model.py to utilities as it is pretty fundamental and should be part of the codebase for debugging purposes.

Reply:
I don't think the same. We already have a boring model for our tests and a boring model for bug reports and debugging. I vote for not including it in utilities.

Reply:
@carmocca Any thoughts ?

Reply:
So I ran into the same issue in https://github.com/PyTorchLightning/pytorch-lightning/pull/7614#discussion_r671183652 (blocked for the same reason).

Some `pl_examples` rely on the `MNISTDataModule`, which is in the test directory.

We currently include everything in our distribution, which is what I try to avoid in the linked PR.

However, CI fails because then the pl_examples do not have access to the `MNISTDataModule` implementation if we do.

So we have two real options:

1. Duplicate the `BoringModel`/`MNISTDataModule`/ implementations in the `pl_examples` directory
2. Move `BoringModel`/`MNISTDataModule` to the source tree (what this PR does) to avoid code duplication.

I think I prefer (2). If you guys agree I can do this change in a separate PR

cc @Borda

Reply:
Lets continue discussing in https://github.com/PyTorchLightning/pytorch-lightning/pull/8776